### ISSUE TITLE
Questions about seed

### ISSUE THREAD

**Issue by SeunghanYu** [gmberton/deep-visual-geo-localization-benchmark#29]
Hello, @gmberton!

I have a question regarding the use of seeds for ensuring deterministic behavior in experiments.

```python
def make_deterministic(seed=0):
    """Make results deterministic. If seed == -1, do not make deterministic.
    Running the script in a deterministic way might slow it down.
    """
    if seed == -1:
        return
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = False
```

Given that I set the seed to 0 and keep all other experimental conditions constant, is it possible for the results to vary across multiple runs of the experiment using the current code?

Additionally, I’m curious whether there would be any difference in the results when running an experiment for 100 epochs straight versus running it for 50 epochs, pausing, and then resuming from epoch 50 to epoch 100.

Any insights on how the seed might influence the outcomes in these scenarios would be greatly appreciated!

Thank you!

**Comment by gmberton** [comment]
Hi, given that everything is seeded the results won't change across experiments if they are run with the same hyperparameters. Some exceptions might exist (e.g. sklearn PCA computation is multi-processed and might change even if the seed is the same).
However, running an experiment for 100 epochs will give (slightly) different results than running it for 50 epochs and then resuming for 50 more epochs, because we don't resume the random generator state.

**Comment by SeunghanYu** [comment]
Thank you so much for your detailed response!
